aws/resource_aws_dynamodb_table.go

```diff
@@ -619,6 +623,30 @@ func resourceAwsDynamoDbTableDelete(d *schema.ResourceData, meta interface{}) er
 
 	log.Printf("[DEBUG] DynamoDB delete table: %s", d.Id())
 
+	if backupName, ok := d.GetOk("final_backup_identifier"); ok {
+		backupInput := &dynamodb.CreateBackupInput{
+			BackupName: aws.String(backupName.(string)),
+			TableName:  aws.String(d.Id()),
+		}
+		err := resource.Retry(10*time.Minute, func() *resource.RetryError {
```

Comment:
This may need to configurable if the size of the table determines the length of the backup operation.